src/test/java/github/weichware10/util/config/ConfigClientTest.java

```diff
@@ -53,7 +56,70 @@ public void loadingShouldHaveCorrectEffect() {
-            client.getConfig(),
-            instanceOf(Configuration.class));
+                client.getConfig(),
+                instanceOf(Configuration.class));
+    }
+
+    /**
+     * Stellt sicher, dass das Speichern nicht möglich ist,
+     * wenn keine Konfiguration geladen wurde.
+     */
+    @Test
+    public void shouldNotWriteEmptyConfig() {
+        ConfigClient client = new ConfigClient();
+        assertFalse("Client should not export before loading", client.writeToJson("config.json"));
+    }
+
+    /**
+     * Stellt sicher, dass nicht an einen invaliden Pfad geschrieben wird.
+     */
+    @Test
+    public void shouldNotWriteToWrongPath() {
+        ConfigClient client = new ConfigClient();
+        client.loadFromJson("src/test/resources/testconfig-CODECHARTS.json");
+        assertFalse("should not write to invalid path", client.writeToJson("badpath/config.json"));
+        assertFalse("should not write to invalid path", client.writeToJson("config"));
+    }
+
+    /**
+     * Testet, ob der ConfigClient eine geladene Konfiguration abspeichern kann.
+     */
+    @Test
+    public void shouldSaveToValidPath() {
+        ConfigClient client = new ConfigClient();
+
+        client.loadFromJson("src/test/resources/testconfig-CODECHARTS.json");
+        assertTrue("Should be able to write to target/testconfig-CODECHARTS.json",
+                client.writeToJson("target/testconfig-CODECHARTS.json"));
+
+        client.loadFromJson("src/test/resources/testconfig-EYETRACKING.json");
+        assertTrue("Should be able to write to target/testconfig-EYETRACKING.json",
+                client.writeToJson("target/testconfig-EYETRACKING.json"));
+
+        client.loadFromJson("src/test/resources/testconfig-ZOOMMAPS.json");
+        assertTrue("Should be able to write to target/testconfig-ZOOMMAPS.json",
+                client.writeToJson("target/testconfig-ZOOMMAPS.json"));
+    }
+
+    /**
+     * Testet, ob die gleiche Konfiguration geladen wird,
+     * wenn zwei Mal die gleiche Datei geladen wird.
+     */
+    @Test
+    public void loadingShouldBeReliable() {
+        ConfigClient client1 = new ConfigClient();
+        ConfigClient client2 = new ConfigClient();
+
+        client1.loadFromJson("src/test/resources/testconfig-CODECHARTS.json");
+        client2.loadFromJson("src/test/resources/testconfig-CODECHARTS.json");
+        assertEquals("Configs should be the same", client1.getConfig(), client2.getConfig());
+
+        client1.loadFromJson("src/test/resources/testconfig-EYETRACKING.json");
+        client2.loadFromJson("src/test/resources/testconfig-EYETRACKING.json");
+        assertEquals("Configs should be the same", client1.getConfig(), client2.getConfig());
+
+        client1.loadFromJson("src/test/resources/testconfig-ZOOMMAPS.json");
+        client2.loadFromJson("src/test/resources/testconfig-ZOOMMAPS.json");
+        assertEquals("Configs should be the same" ,client1.getConfig(), client2.getConfig());
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>',' is not followed by whitespace.